### PR TITLE
generate at breakpoints handles complex selectors

### PR DESCRIPTION
**Pull request by AlecRaeside**
When passing in a complex selector with a `>` you need to put it in quotes as sass will try to evaluate it as a mathematical equation

### CODE REVIEW

**Comment by AlecRaeside** [comment]
@chris-pearce @kllevin I've included 2 examples for the list inline helper, one with a simple selector and one with a complex selector. Will complete the file once this is LGTMed.

Changes in usage for the mixin from before is it no longer puts a `.` at the start for you, so it must be a full selector (ie you may not put it a basic class so it shouldnt do it for you). This leading dot means it always has to be a string in quotes

**Comment by kllevin** [comment]
LGTM

File: utilities/_u-list-inline.scss [hunk 1]
```diff
@@ -189,12 +189,20 @@ $list-inline-divider-thickness:    1 !default;
...
+  content: "/";
+}
+
   // When used in conjuction with the Spacing -> Base modifier
 
   // Make the pseudo element `inline-block` as it needs `margin-right`
   %u-list-inline--delimited-slash[class*="u-list-inline--spacing"] > li + li:before,
   .u-list-inline--delimited-slash[class*="u-list-inline--spacing"] > li + li:before {display: inline-block;}
 
+  @include generate-at-breakpoints('.u-list-inline--delimited-slash[bp][class*="u-list-inline--spacing"] > li + li:before', palm lap desk) {
```
Comment (AlecRaeside) [inline utilities/_u-list-inline.scss]: so thats the one downside of using this placeholder, to the untrained eye it looks like im selecting an element with the attribute `bp="something"`. I'll try uding curly brackets to see if they look better

**Comment by AlecRaeside** [comment]
I've switched to using curly braces for the placeholder, not it looks more sassy and doesn't clash with attribute selectors

File: core/mixins/_generate-at-breakpoints.scss [hunk 2]
```diff
@@ -10,29 +10,19 @@
...
 
-   @include generate-at-breakpoints(u-align-v-inline-top, palm lap, false) {
+   @include generate-at-breakpoints('.u-align-v-inline-top[bp] li', palm lap, false) {
      vertical-align: top;
    }
  */
 
 
-@mixin generate-at-breakpoints($class, $breakpoint-names: (), $placeholder: true) {
+$mixin-breakpoint-name-placeholder: "{bp}";
```
Comment (chris-pearce) [inline core/mixins/_generate-at-breakpoints.scss:28]: Can we add a comment about this?

File: core/mixins/_generate-at-breakpoints.scss [hunk 3]
```diff
@@ -43,10 +33,18 @@
...
-    @include respond-to($breakpoint-name, $limit: if($breakpoint-name ==
-      'palm', 'max', 'min')) {
+    $limit: if($breakpoint-name == 'palm', 'max', 'min');
+
+    @include respond-to($breakpoint-name, $limit) {
+      $final-selector: "#{$class}-at-#{$breakpoint-name}";
+
+      // Handle complex selectors by replacing a placeholder with the breakpoint suffix
+      @if str_index($class, $mixin-breakpoint-name-placeholder) != null {
+        $final-selector: str-replace($class, $mixin-breakpoint-name-placeholder, "-at-#{$breakpoint-name}");
```
Comment (chris-pearce) [inline core/mixins/_generate-at-breakpoints.scss]: We need to wrap at 80 chars for Scally and going forward for the new CSS Guidelines. I've got this in my Sublime setting file:

```
"word_wrap": true,
"wrap_width": 80
```

**Pull request merged**